spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/FlywayMigrationScriptMissingFailureAnalyzer.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.diagnostics.analyzer;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.springframework.boot.diagnostics.AbstractFailureAnalyzer;
+import org.springframework.boot.diagnostics.FailureAnalysis;
+
+/**
+ * A {@code FailureAnalyzer} that performs analysis of failures caused by a
+ * {@code IllegalStateException}.
```

Comment:
We can't really do that. The purpose of a failure analyzer is to tracker a specific exception. This is a sign that we need to provide such exception if it doesn't exist yet to avoid magic string  comparison.

Reply:
I understand. I have created a new exception as advised.